test_expressions/test_expressions.cc

```diff
@@ -121,10 +121,9 @@ void Test1() {
 
 template <typename D>
 void Test2() {
-  auto c1 = D::C(2);
-  auto e1 = D::BOp('+', c1, c1);
```

Comment:
the definition of D::BOp does std::move on each of its arguments.
this code is invalid.
i am surprised clang-tidy does not catch this for you
and i am double surprised that your code compiles and runs it.

Reply:
Where is that definition specified? In fact, KyExpressionsDriver has:
`static Expr BinaryOperatorExpression(char op, Expr l, Expr r)`
Are you looking for l and r to be rvalue references instead of copy by value?

On clang-tidy comments:
1. Which clang-tidy check was expected to catch which behavior?
2. My code works because I added a copy constructor, and I assumed you wanted one based on your requested interface. Can you update the interface if you meant something else (or alternately let me know if there is a different convention/requirement implied)?